src/npctrade.cpp

```diff
@@ -163,6 +163,10 @@ std::vector<item_pricing> npc_trading::init_buying( player &buyer, player &selle
             return;
         }
 
+        if( it.count() <= 0 ) {
```

Comment:
This appears like it's hiding a bug. If `item::count` returns 0, that means it's an item counted by charges, but with 0 charges. For example "water (0)", which makes no sense. There should not be such an item.

I suggest you add some debug message about this inconsistent state.